### PR TITLE
Deprecate JedisPoolAbstract

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

File: src/main/java/redis/clients/jedis/JedisPool.java [hunk 1]
```diff
@@ -333,21 +330,20 @@ public Jedis getResource() {
...
   @Override
-  protected void returnResource(final Jedis resource) {
+  public void returnResource(final Jedis resource) {
     if (resource != null) {
       try {
         resource.resetState();
         returnResourceObject(resource);
       } catch (Exception e) {
         returnBrokenResource(resource);
-        throw new JedisException("Resource is returned to the pool as broken", e);
```
Comment (yangbodong22011) [inline src/main/java/redis/clients/jedis/JedisPool.java:350]: I agree not to throw a `JedisException`.
but we should logger.error for it (normal resource became to broken)

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/JedisPool.java:350]: acknowledged

File: src/main/java/redis/clients/jedis/JedisPool.java [hunk 2]
```diff
@@ -333,21 +332,14 @@ public Jedis getResource() {
...
+        log.debug("Resource is returned to the pool as broken", e);
```
Comment (yangbodong22011) [inline src/main/java/redis/clients/jedis/JedisPool.java:342]: I think we should use `log.warn` or `log.error`, `log.debug` is generally not used in a production environment, but we need to record errors.

Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisPool.java:342]:
```suggestion
        log.warn("Resource is returned to the pool as broken", e);
```

**Review by yangbodong22011 — APPROVED** [review]

**Review by gkorland — APPROVED** [review]

**Pull request merged**